### ISSUE TITLE
DTB- Invalid value not in range

### ISSUE THREAD

**Issue by moda20** [NiKoTron/dart-tags#24]
**Describe the bug**
a bug when trying to get the comments out of a file after writing new tags in, that includes new comments

**Environment**
- Platform Flutter
- Library version 0.3.1
- Tag type ID3 2.4

**To Reproduce**
execute this code : 
```
 final com1 = Comment('eng', 'desc_here', 'lol it is a comment');
      final com2 = Comment('eng', 'desc_here_2', 'lol it is a comment');
      final pic = AttachedPicture(
          'image/jpeg', 0x03, 'foo.jpg', picture.readAsBytesSync());
      final tag = Tag()
        ..tags = {
          'title': 'foo',
          'artist': 'bar',
          'album': 'baz',
          'year': '2010',
          'comment': {
            com1.key: com1,
            com2.key: com2,
          },
          'track': '6',
          'genre': 'Dream',
          'custom': 'Just a tag',
       // No picture used
        }
        ..type = 'ID3'
        ..version = '2.4';

      final writer = ID3V2Writer();
      final blocks = writer.write(await file2.readAsBytes(), tag);

      final r = ID3V2Reader();
      final f = await r.read(blocks);

```

**Additional context**
the full error 

```
[ERROR:flutter/lib/ui/ui_dart_state.cc(166)] Unhandled Exception: RangeError (end): Invalid value: Not in inclusive range 12..30: -1
E/flutter (14192): #0      RangeError.checkValidRange (dart:core/errors.dart:357:9)
E/flutter (14192): #1      List.sublist (dart:core-patch/growable_array.dart:84:38)
E/flutter (14192): #2      ApicFrame.decodeBody (package:dart_tags/src/frames/id3v2/apic_frame.dart:79:25)
E/flutter (14192): #3      ID3V2Frame.decode (package:dart_tags/src/frames/id3v2/id3v2_frame.dart:57:39)
E/flutter (14192): #4      ID3V2Reader.parseValues (package:dart_tags/src/readers/id3v2.dart:71:23)
E/flutter (14192): #5      Reader.read.<anonymous closure> (package:dart_tags/src/readers/reader.dart:13:23)
E/flutter (14192): #6      _rootRunUnary (dart:async/zone.dart:1198:47)
E/flutter (14192): #7      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
E/flutter (14192): #8      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
E/flutter (14192): #9      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
E/flutter (14192): #10     Future._propagateToListeners (dart:async/future_impl.dart:725:32)
E/flutter (14192): #11     Future._completeWithValue (dart:async/future_impl.dart:529:5)
E/flutter (14192): #12     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:40:15)
```

**Comment by NiKoTron** [comment]
Hi! @moda20, Sorry for late answer I was on vaction. I found that you opened a few issues and than close it. Could you please describe is issues still valid or not?

**Comment by moda20** [comment]
I had some issues with the specific tag types, like I needed to know that I can't put int in year and track tags but rather strings instead. that flipped me over a lot. there are some other things like old tests in previous issues that I referred to without checking the commit date. this issue in particular is when I used an empty list for the picture bytes when instantiating `AttachedPicture`. ( i tried this because I didn't know how to pass null as a picture, which later on I knew that I just needed to not pass it in the tags map)
I found my way through all my problems and the library is fine. But I think it needs a bit more documentation, like a README file rather than referring to the test folder. 

I'll be happy to do it but I am quite busy right now. That's it.

Thanks for reaching back.
